### PR TITLE
feat: detect finding status updates from PR comment replies (#60)

### LINKED ISSUE #60: feat: detect finding status updates from replies and reactions
## Parent Issue
Part of #53 - Finding deduplication and status tracking

## Summary
Allow authors to update finding status by replying to inline comments or reacting to the tracking comment.

### 1. Reply to Inline Comment
```
Bot: **HIGH** SQL injection in db.go:45...

Author replies: "Acknowledged - this is intentional for admin-only endpoint"
```
→ Parse reply, update finding status to `acknowledged` with reason

### 2. Keywords in Replies
Detect status-change keywords:
- "acknowledged", "won't fix", "intentional" → `acknowledged`
- "disputed", "false positive", "disagree" → `disputed`
- "fixed", "resolved" → `resolved` (though auto-detect is better)

### 3. Reactions (Future Enhancement)
- 👍 on tracking comment finding → acknowledged
- 👎 on tracking comment finding → disputed

## Implementation

```go
type StatusUpdateDetector interface {
    // Check inline comment replies for status updates
    DetectFromReplies(ctx context.Context, owner, repo string, prNumber int) ([]StatusUpdate, error)
}

type StatusUpdate struct {
    FindingFingerprint FindingFingerprint
    NewStatus          FindingStatus
    Reason             string
    UpdatedBy          string
    UpdatedAt          time.Time
}
```

## Files to create
- `internal/adapter/github/status_detector.go`

## Acceptance Criteria
- [ ] Can parse inline comment threads
- [ ] Detects acknowledgment keywords
- [ ] Detects dispute keywords
- [ ] Returns structured status updates
- [ ] Handles multiple findings in same thread

### PR DESCRIPTION
**Pull request by bkyoung**
## Summary

- Enables users to acknowledge or dispute code review findings by replying to inline comments with status keywords
- Scans PR comments, matches replies to original findings via embedded fingerprints, and updates finding statuses in the tracking system
- Part of Epic #53 (Finding Deduplication and Status Tracking)

### Changes

**Domain layer:**
- `status_keywords.go`: Parses status keywords ("acknowledged", "disputed", "false positive", etc.) with word boundary detection to prevent false positives

**Adapter layer (GitHub):**
- `request_builder.go`: Embeds fingerprints in inline comments using HTML comment markers (`<!-- CR_FINGERPRINT:... -->`)
- `reply_fetcher.go`: Fetches PR review comments with pagination support
- `status_detector.go`: Matches replies to findings and detects status updates
- `status_scanner.go`: Implements `StatusScanner` interface for platform-agnostic status detection

**Usecase layer:**
- `orchestrator.go`: Adds `StatusScanner` interface and integrates status scanning into review flow (replaces TODO(#60))

**Wiring:**
- `main.go`: Wires up the `GitHubStatusScanner` when GitHub posting is enabled

### Supported Keywords

| Action | Keywords |
|--------|----------|
| Acknowledge | `acknowledged`, `won't fix`, `wont fix`, `intentional`, `ack` |
| Dispute | `disputed`, `false positive`, `not a bug`, `not an issue` |

## Test plan

- [x] Unit tests for keyword parsing with edge cases (word boundaries, case insensitivity)
- [x] Unit tests for fingerprint embedding/extraction
- [x] Unit tests for reply fetching with pagination
- [x] Unit tests for status detection logic
- [x] Unit tests for status scanner integration
- [x] All tests pass: `go test ./...`
- [x] Build succeeds: `go build -o cr ./cmd/cr`

Closes #60

### CODE REVIEW

File: internal/adapter/github/reply_fetcher.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	visitedURLs := make(map[string]bool) // Prevent infinite pagination loops
+	pageCount := 0
+
+	// Start with the first page
+	nextURL := fmt.Sprintf("%s/repos/%s/%s/pulls/%d/comments?per_page=100",
+		c.baseURL, url.PathEscape(owner), url.PathEscape(repo), pullNumber)
+
+	for nextURL != "" {
+		// Pagination loop protection
+		if pageCount >= maxPaginationPages {
```
[the remark this answers is not in the document]
Comment (bkyoung) [reply internal/adapter/github/reply_fetcher.go]: false positive - maxPaginationPages constant is defined in client.go (line 24) as const maxPaginationPages = 100. Same package, different file.

File: internal/adapter/github/reply_fetcher.go [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+		return nil, "", err
+	}
+	defer resp.Body.Close()
+
+	var comments []PullRequestComment
+	if err := json.NewDecoder(resp.Body).Decode(&comments); err != nil {
+		return nil, "", fmt.Errorf("failed to parse response: %w", err)
+	}
+
+	// Parse Link header for pagination
```
[the remark this answers is not in the document]
Comment (bkyoung) [reply internal/adapter/github/reply_fetcher.go]: false positive - parseNextLink is defined in client.go (lines 401-413). The function is accessible from reply_fetcher.go since they share the same package.

File: internal/domain/status_keywords.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	textLower := strings.ToLower(text)
+
+	// Check acknowledge keywords first (they're more common)
+	for _, keyword := range AcknowledgeKeywords {
+		if containsKeyword(textLower, keyword) {
+			return FindingStatusAcknowledged, truncateReason(text), true
+		}
+	}
+
+	// Check dispute keywords
```
[the remark this answers is not in the document]
Comment (bkyoung) [reply internal/domain/status_keywords.go]: acknowledged - This is a valid bug. The code only checks the first occurrence of a keyword. Will fix by iterating through all occurrences until a valid word-boundary match is found.

**Comment by bkyoung** [comment]
## Review Cycle Summary

This PR went through 2 review cycles with the following outcomes:

### Status Update Detection (Testing #60 Feature)
The new status detection feature worked correctly:
- ✅ **7 findings auto-resolved** by code fixes
- 💬 **1 finding acknowledged** then fixed
- ⚠️ **4 findings disputed** as false positives (functions defined in same package)

### Findings Addressed
1. **Fixed:** Keyword matching now scans all occurrences (not just first)
2. **Fixed:** Added fingerprint format validation (32-char hex)
3. **Disputed:** 4 high-severity findings about "missing" functions that are defined in `client.go` (same package)

### Remaining Open Findings
- **1 High (false positive):** "Infinite loop vulnerability" - the loop correctly terminates via `searchStart` increment and bounds checking
- **14 Medium:** Mostly suggestions (performance, maintainability, test coverage) - acceptable for MVP
- **15 Low:** Minor improvements for future iterations

### Conclusion
No valid critical/high bugs remain. The feature implementation is complete and tested.
